### DISCUSSION TITLE
Changing background of particles canvas onchange for input.

### DISCUSSION THREAD

**Discussion by abhishekkokate** [matteobruni/tsparticles#3310] [Q&A] [answered]
I am using react-tsparticles package in my react project. I want to change the canvas background with the value in input text.

I tried changing the values for my options object but it didn't work. is there any way i can make it work?

```
var bgColor = "#0c0032";
const particlesOptions = {
   --------- Options ------------
}

```

`<Particles id="tsparticles" options={particlesOptions}  init={particlesInit} loaded={particlesLoaded}  />`

`<input
            type="text"
            placeholder="Background color"
            defaultValue={bgColor}
            onChange={(e) => {
              bgColor = e.target.value;
            }}
            ref={BgColorValue}
          />`

**Comment by matteobruni** [comment, marked as answer]
Hi @abhishekkokate,

I think that `bgColor` should be a `state` variable, not a classic one.
